### PR TITLE
Tic tac toe game

### PR DESCRIPTION
**Pull request by JozefR**
Be kind !

### CODE REVIEW

File: my-app/src/index1.js [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+  }
+
+  handleClick(i) {
+    if (this.state.winner != null) {
+      return;
+    }
+
+    const squares = this.state.squares.slice();
+    const squaresHistory = this.state.squaresHistory.slice();
+    const xIsNext = !this.state.xIsNext;
```
Comment (juliavorcakova) [inline my-app/src/index1.js:24]: I don't understand.. why falsy? In my understanding xIsNext should be just xIsNext = this.state.xIsNext;
Also you dont need to assing it to new const you can just use directly this.state.xIsNext;

Comment (juliavorcakova) [reply my-app/src/index1.js:24]: Ahhh I understand now - you want to toggle it right?

File: my-app/src/index1.js [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+      player: player,
+      squares: squares,
+    };
+
+    squaresHistory.push(history);
+
+    const winner = calculateWinner(squares);
+
+    this.setState({
+      squares: squares,
```
Comment (juliavorcakova) [inline my-app/src/index1.js]: Just so you know, easier syntax is just {squares, swaureHistory...} - if the key is named the same as the value ocnst u can use just the above syntax.

File: my-app/src/index1.js [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+  }
+
+  return <div className="status">{playerStatus}</div>;
+}
+
+function ChossedOptions(props) {
+  const historyOptions = props.squaresHistory.slice();
+
+  let index = -1;
+  var historyElements = historyOptions.map((element) => {
```
Comment (juliavorcakova) [inline my-app/src/index1.js:91]: You don't have to handle the index separately - map has index as a second argument so u can just do (element, index) => ...
Also try not to use 'var' and just return it directly, you dont have to assign it to const and then return it ;)

File: my-app/src/index1.js [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+  });
+
+  return historyElements;
+}
+
+function HistoryOption(props) {
+  let buttonText = "";
+
+  if (props.index === 0) {
+    buttonText += "Go to game start";
```
Comment (juliavorcakova) [inline my-app/src/index1.js]: why += just =

File: my-app/src/index1.js [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+      </p>
+    </div>
+  );
+}
+
+function RenderSquareHistory(props) {
+  const value = props.squares.squares[props.index];
+
+  if (value !== null) {
+    return <Square value={value} onClick={props.onClick} />;
```
Comment (juliavorcakova) [inline my-app/src/index1.js]: To simplify it you can just do `return <Square value={value || ''} onClick={props.onClick} />`. The || takes the first value if it's not falsy othervise it takes the second one. Or if you dont like it just fo `return <Square value={value !== null ? value : ''} onClick={props.onClick} />;`

File: my-app/src/index1.js [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+  } else {
+    return <Square value="" onClick={props.onClick} />;
+  }
+}
+
+function RenderSquare(props) {
```
Comment (juliavorcakova) [inline my-app/src/index1.js]: I dont understand why u have this as a separate component - you should avoid having component which just perform some action. Think of component as UI units (you can really see them in UI). The same for others 'Render...' components.

File: my-app/src/index1.js [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+          })}
+        </div>
+      </div>
+    );
+  }
+
+  return (
+    <div>
+      <div className="board-row">
+        <RenderSquare
```
Comment (juliavorcakova) [inline my-app/src/index1.js]: Just call directly <Square> component, why the abstraction

File: my-app/src/index1.js [hunk 8]
```diff
@@ -0,0 +1,291 @@
...
+          onClick={(index) => props.onClick(props.index)}
+        >
+          {buttonText}
+        </button>
+      </p>
+    </div>
+  );
+}
+
+function RenderSquareHistory(props) {
```
Comment (juliavorcakova) [inline my-app/src/index1.js]: Same as below. This component is not necesary. If you need to perform some conditions and you don't want to do it directly at the html (jsx) code then just create a function in <Board> component, but not a new component just to render another component.

File: my-app/src/index1.js [hunk 9]
```diff
@@ -0,0 +1,291 @@
...
+    <Square
+      value={props.squares[props.index]}
+      onClick={(index) => props.onClick(props.index)}
+    />
+  );
+}
+
+function Square(props) {
+  return (
+    <button className="square" onClick={props.onClick}>
```
Comment (juliavorcakova) [inline my-app/src/index1.js:133]: Wait so at the end you are not sending the index as a param of the onclick. So where is the index param in <Game> compoent 'handleClick' fn coming from?

**Review by juliavorcakova — CHANGES_REQUESTED** [review]
You rocked that!

File: my-app/src/index1.js (hunk 1 above)
Comment (JozefR) [reply my-app/src/index1.js:24]: Yes it's toggled every time user clicks the square button.

File: my-app/src/index1.js (hunk 2 above)
Comment (JozefR) [reply my-app/src/index1.js]: Thanks, i will fix it.

**Pull request merged**